presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -215,11 +216,43 @@ private synchronized Path getNextSpillPath()
     private boolean hasEnoughDiskSpace(Path path)
     {
         try {
-            FileStore fileStore = getFileStore(path);
-            return fileStore.getUsableSpace() > fileStore.getTotalSpace() * (1.0 - maxUsedSpaceThreshold);
+            HetuFileSystemClient fileSystemClient = getFileSystem(path, spillToHdfs, spillProfile, fileSystemClientManager);
+            return fileSystemClient.getUsableSpace(path) > fileSystemClient.getTotalSpace(path) * (1.0 - maxUsedSpaceThreshold);
         }
         catch (IOException e) {
             throw new PrestoException(OUT_OF_SPILL_SPACE, "Cannot determine free space for spill", e);
         }
     }
+
+    private void createSpillDirectories()
```

Comment:
should we not check that if the profile is hdfs, there should be only 1 spiller path? or are we supporting multiple hdfs spiller paths?

Reply:
We are supporting multiple spill paths only in case of disk spilling , in case of hdfs we are supporting only one path.
It is being already taken care from the features config..
![Screenshot from 2022-03-02 10-31-22](https://user-images.githubusercontent.com/99017840/156298416-f8b908ed-034b-476c-9894-f21f644378c9.png)